### PR TITLE
Forward test for refactoring

### PR DESCRIPTION
**Pull request by xinliupitt**
# Description

> :memo: Please include a summary of the change. 
>  
> * This pull request adds a forward pass test file to guard the correctness of refactoring Transformer model. The original Transformer model is in nlp/transformer/transformer.py. The refactored model is in nlp/modeling/models/seq2seq_transformer.py. This forward pass test file can test whether the original model and refactored model have the same output after one forward pass, given that:
> ** Inputs are the same.
> ** Weights are the same.
> ** Dropout randomness is avoided.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x] TensorFlow 2 migration

## Tests

> :memo: Please describe the tests that you ran to verify your changes.
>  
> * Test cases in this file can be passed. 

**Test Configuration**:

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [ ] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [ ] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [ ] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: official/nlp/transformer/transformer_forward_test.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    # src_model is the original model before refactored.
+    src_model = transformer.create_model(self.params, True)
+    src_num_weights = _count_params(src_model)
+    src_weights = src_model.get_weights()
+    src_model_output = src_model([inputs, targets], training=True)
+
+    # dest_model is the refactored model.
+    dest_model = seq2seq_transformer.create_model(self.params, True)
+    dest_num_weights = _count_params(dest_model)
+    if src_num_weights != dest_num_weights:
```
Comment (saberkun) [inline official/nlp/transformer/transformer_forward_test.py:57]: can you use assertEqual here?

File: official/nlp/transformer/transformer_forward_test.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      raise ValueError("Source weights can't be set to destination model due to"
+                       "different number of weights.")
+    dest_model.set_weights(src_weights)
+    dest_model_output = dest_model([inputs], training=False)
+    self.assertAllEqual(src_model_output[0], dest_model_output[0])
+    self.assertAllEqual(src_model_output[1], dest_model_output[1])
+
+def _count_params(layer, trainable_only=True):
```
Comment (saberkun) [inline official/nlp/transformer/transformer_forward_test.py:86]: Move it to top
